crates/evm/src/rpc_helpers/filter.rs

```diff
@@ -115,3 +131,290 @@ pub fn convert_block_number(
     };
     Ok(Some(num))
 }
+
+/// All active filters
+#[derive(Debug, Clone, Default)]
+pub struct ActiveFilters {
+    inner: Arc<Mutex<HashMap<FilterId, ActiveFilter>>>,
+}
+
+impl ActiveFilters {
+    /// Returns an empty instance.
+    pub fn new() -> Self {
+        Self {
+            inner: Arc::new(Mutex::new(HashMap::default())),
+        }
+    }
+}
+
+/// An installed filter
+#[derive(Debug)]
+struct ActiveFilter {
+    /// At which block the filter was polled last.
+    block: u64,
+    /// Last time this filter was polled.
+    last_poll_timestamp: Instant,
+    /// What kind of filter it is.
+    kind: FilterKind,
+}
+
+#[derive(Clone, Debug)]
+/// The kind of filter
+pub enum FilterKind {
+    /// Log filter
+    Log(Box<Filter>),
+    /// Block filter
+    Block,
+    /// Pending transaction filters are not supported
+    /// and will return unsupported error if used.
+    PendingTransaction,
+}
+
+/// Idea from: https://github.com/paradigmxyz/reth/blob/ed7da87da4de340a437bf46f39a7e1397ac82065/crates/rpc/rpc/src/eth/filter.rs#L382
+#[derive(Clone)]
+pub struct CitreaFilter {
+    /// All currently installed filters.
+    pub active_filters: ActiveFilters,
+    /// Provides ids to identify filters
+    pub id_provider: Arc<dyn IdProvider>,
+    /// Task executor to spawn the stale filter clearing task
+    pub task_executor: TaskExecutor,
+    /// Duration since the last filter poll, after which the filter is considered stale
+    pub stale_filter_ttl: Duration,
+}
+
+impl CitreaFilter {
+    /// Creates a new instance of the CitreaFilter.
+    pub fn new(
+        task_executor: reth_tasks::TaskExecutor,
+        stale_filter_ttl: Option<usize>,
+    ) -> CitreaFilter {
+        let citrea_filter = CitreaFilter {
+            active_filters: ActiveFilters::new(),
+            id_provider: Arc::new(EthSubscriptionIdProvider::default()),
+            task_executor,
+            stale_filter_ttl: stale_filter_ttl
+                .map(|d| Duration::from_secs(d as u64))
+                .unwrap_or_else(|| DEFAULT_STALE_FILTER_TTL),
+        };
+
+        let this = citrea_filter.clone();
+        tracing::info!("AAStarting stale filter clearing task ");
```

Comment:
```suggestion
        tracing::info!("Starting stale filter clearing task ");
```

Should this be info?

Reply:
No, forgot to change it to trace